src/EA.Weee.Web.Tests.Unit/Areas/Aatf/Requests/EditEvidenceNoteRequestCreatorTests.cs

```diff
@@ -0,0 +1,283 @@
+            model.ReceivedId = receivedId;
+            model.WasteTypeValue = wasteTypeValue;
+            model.ProtocolValue = protocolValue;
+            model.CategoryValues = tonnageValues;
+            model.Id = id;
+            model.Action = ActionEnum.Save;
+
+            //act
+            var request = requestCreator.ViewModelToRequest(model);
+
+            //assert
+            request.Id.Should().Be(id);
+            request.AatfId.Should().Be(aatfId);
+            request.OrganisationId.Should().Be(organisationId);
+            request.StartDate.Should().Be(startDate);
+            request.EndDate.Should().Be(endDate);
+            request.RecipientId.Should().Be(receivedId);
+            request.WasteType.Should().Be(wasteTypeValue);
+            request.Protocol.Should().Be(protocolValue);
+            request.Status.Should().Be(NoteStatus.Draft);
+            ShouldContainTonnage(tonnageValues, request);
+        }
+
+        [Fact]
+        public void ViewModelToRequest_GivenSubmitViewModel_CreateEvidenceNoteRequestShouldBeCreated()
+        {
+            //arrange
+            var id = fixture.Create<Guid>();
+            var aatfId = fixture.Create<Guid>();
+            var endDate = fixture.Create<DateTime>();
+            var startDate = fixture.Create<DateTime>();
+            var organisationId = fixture.Create<Guid>();
+            var receivedId = fixture.Create<Guid>();
+            var wasteTypeValue = fixture.Create<WasteType>();
+            var protocolValue = fixture.Create<Protocol>();
+            var tonnageValues = fixture.CreateMany<EvidenceCategoryValue>()
+                .ToList();
+
+            var model = ValidModel();
+            model.AatfId = aatfId;
+            model.EndDate = endDate;
+            model.StartDate = startDate;
+            model.OrganisationId = organisationId;
+            model.ReceivedId = receivedId;
+            model.WasteTypeValue = wasteTypeValue;
+            model.ProtocolValue = protocolValue;
+            model.CategoryValues = tonnageValues;
+            model.Id = id;
+            model.Action = ActionEnum.Submit;
+
+            //act
+            var request = requestCreator.ViewModelToRequest(model);
+
+            //assert
+            request.Id.Should().Be(id);
+            request.AatfId.Should().Be(aatfId);
+            request.OrganisationId.Should().Be(organisationId);
+            request.StartDate.Should().Be(startDate);
+            request.EndDate.Should().Be(endDate);
+            request.RecipientId.Should().Be(receivedId);
+            request.WasteType.Should().Be(wasteTypeValue);
+            request.Protocol.Should().Be(protocolValue);
+            request.Status.Should().Be(NoteStatus.Submitted);
+            ShouldContainTonnage(tonnageValues, request);
+        }
+
+        [Fact]
+        public void ViewModelToRequest_GivenViewModelWithNullValues_CreateEvidenceNoteRequestShouldBeCreated()
```

Comment:
..WithNullValies?

Reply:
Will rename the test, as should be GivenViewModelWithNullWasteAndProtocolValues